File: C#/263_Ugly_Number.cs
using System;

namespace LeetCode
{
    public class Solution
    {
        public bool IsUgly(int num)
        {
            if (num <= 0)
                return false;
            int[] arr_factors = { 2, 3, 5 };
            foreach (int elem in arr_factors)
            {
                while (num % elem == 0)
                {
                    num /= elem;
                }
            }
            return num == 1;
        }

        //Testcase
        static void Main(string[] args)
        {
            Solution s = new Solution();
            int rand_num = new Random().Next(0, 11);
            if (s.IsUgly(rand_num))
                Console.Write(rand_num + " is a ugly number.");
            else
                Console.Write(rand_num + " isn't a ugly number.");
        }
    }
}

File: Python/263_Ugly_Number.py
import random


class Solution(object):
    def isUgly(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num <= 0:
            return False
        for x in [2, 3, 5]:
            while num % x == 0:
                num /= x
        return num == 1


def test():
    s = Solution()
    rand_num = random.randint(0, 10)
    if s.isUgly(rand_num):
        print str(rand_num) + " is a ugly number."
    else:
        print str(rand_num) + " isn't a ugly number."


if __name__ == '__main__':
    test()